symfony/src/Controller/Management/Structure/PrefilledAnswersController.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace App\Controller\Management\Structure;
+
+
+
+use App\Base\BaseController;
+use App\Entity\PrefilledAnswers;
+use App\Entity\Structure;
+use App\Form\Type\PrefilledAnswersType;
+use App\Manager\PrefilledAnswersManager;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route(path="management/structures/{structure}/prefilled-answers", name="management_structures_prefilled_answers_")
+ * @ParamConverter("structure", options={"id" = "structure"})
+ * @Security("is_granted('STRUCTURE', structure)")
+ */
+class PrefilledAnswersController extends BaseController
+{
+
+    /**
+     * @var PrefilledAnswersManager
+     */
+    private $prefilledAnswersManager;
+
+    public function __construct(PrefilledAnswersManager $prefilledAnswersManager)
+    {
+        $this->prefilledAnswersManager = $prefilledAnswersManager;
+    }
+
+    /**
+     * @Route("/", name="list")
+     * @Template("management/structures/prefilled_answers/list.html.twig")
+     *
+     * @param Structure $structure
+     *
+     * @return array
+     */
+    public function listPrefilledAnswers(Structure $structure)
+    {
+        $prefilledAnswers = $this->prefilledAnswersManager->getPrefilledAnswersByStructure($structure);
+
+        return ['pager' => $this->getPager($prefilledAnswers), 'structure' => $structure];
+    }
+
+
+    /**
+     * @Route("/{prefilledAnswers}/editor", requirements={"prefilledAnswers" = "\d+"}, name="edit")
+     * @Route("/new", name="new")
+     * @Template("management/structures/prefilled_answers/editor.html.twig")
+     *
+     * @param Structure $structure
+     *
+     * @return array|\Symfony\Component\HttpFoundation\RedirectResponse
```

Comment:
```suggestion
     * @return array|\Symfony\Component\HttpFoundation\RedirectResponse
     *  @IsGranted("STRUCTURE", subject="structure")
```

Reply:
I already done this check but on the class scope and not on the method.

You prefer to do this check on the methods ?

Reply:
oh nice, sorry, never mind